base/src/java/lang/util.d

```diff
@@ -387,31 +387,103 @@ version(Tango){
 
     class Format{
         template UnTypedef(T) {
-            version(Tango){
-                mixin(r"static if (is(T U == typedef))
-                    alias std.traits.Unqual!(U) UnTypedef;
-                else
-                    alias std.traits.Unqual!(T) UnTypedef;");
-            } else { // Phobos
-                alias std.traits.Unqual!(T) UnTypedef;
-            }
+            alias UnTypedef = std.traits.Unqual!(T);
         }
-        static String opCall(A...)( String _fmt, A _args ){
-            //Formatting a typedef is deprecated
+        static String opCall(A...)(in String fmt, A _args) {
+            auto app = std.array.appender!(String)();
             std.typetuple.staticMap!(UnTypedef, A) args;
-            foreach(i, _a; _args) {
-                version(Tango){
-                    mixin(r"static if (is(T U == typedef))
-                        args[i] = cast(U) _a;
-                    else
-                        args[i] = _a;");
-                } else { // Phobos
-                    args[i] = _a;
+            int argIndex = 0;
+            bool increment = false; // should the argIndex be incremented
+
+            // 0 = "basic type", 1 = array, 2 = assocArray (no impl. yet)
+            ubyte[] isRange = new ubyte[_args.length];
+            foreach(i, a; _args) {
+                args[i] = a;
+
+                if (std.traits.isSomeString!(typeof(a))) {
+                    // while strings are a range, we don't want to
+                    // format them differently
+                    isRange[i] = 0;
+                } else if (std.traits.isArray!(typeof(a))) {
+                    // output should be [elem, elem, elem]
+                    isRange[i] = 1;
+                } else if (std.traits.isAssociativeArray!(typeof(a))) {
+                    // output should be {key => value, key2 => value2}
+                    isRange[i] = 2;
+                } else {
+                    // no special formatting by default
+                    isRange[i] = 0;
+                }
+            }
+
+            L: for (size_t i = 0; i < fmt.length; i++) {

```

Comment:
Can `foreach` be used instead.

Reply:
I changed it over to use a `foreach` loop, the commit 58c4718 contains this and the old version commented out which I'll remove if you're happy with the new version.

(I'll squash the commit into 9e3b831)